cmake/templates/python_win32_wrapper.cpp.in

```diff
@@ -87,20 +92,63 @@ int main(int argc, char* argv[]) try
         fprintf(stderr, "[DEBUG] Python script name: %s\n", scriptName.c_str());
 #endif
 
-        // use quoted string to ensure the correct path is used
-        return "\"" + scriptName + "\"";
+        return scriptName;
     };
 
-    const auto GetPythonExecutable = []() -> std::string
+    const auto GetPythonExecutable = [](const std::string& exeFullPath) -> std::string
     {
-        std::string pythonExecutable = "@PYTHON_EXECUTABLE@";
+        std::string scriptName;
+        scriptName += dirnameOf(exeFullPath);
+        scriptName += "\\@ARG_SCRIPT_NAME@";
+
+        std::ifstream infile(scriptName);
+
+        std::string firstLine;
+        if (std::getline(infile, firstLine) && firstLine.find(SHEBANG) == 0)
+        {
+            firstLine.erase(0, SHEBANG.size());
+
+            // spaces & tabs are considered as extra
+            std::string extra_chars(" \t");
+
+            // left trim extra characters.
+            auto ltrim = firstLine.find_first_not_of(extra_chars);
+            while ((std::string::npos != ltrim) &&
```

Comment:
Why should this be a `while` loop? Isn't it being looped through at max once?

Same below.

Reply:
You are right. I misread the effect of the API. Now it is corrected.